scripts/operations/system_recovery/rapid_reinstall.sh

```diff
@@ -0,0 +1,110 @@
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+
+# Print an error message and exit
+function err_exit {
+  echo "ERROR: $*" >&2
+  exit 1
+}
+
+# Must run from m001
+if [[ $(hostname) != "ncn-m001" ]]; then
+  err_exit "Must be run on m001"
+fi
+
+# Yapl must be installed
+rpm -qi yapl > /dev/null 2>&1 || err_exit "Yapl must be installed"
+
+# Check that needed environment variables are set
+env_vars=("CSM_RELEASE" "SYSTEM_NAME")
+for each in "${env_vars[@]}"; do
+  if [ -z "${!each}" ]; then
+    err_exit "$each must be defined"
+  fi
+done
+
+export PITDATA=/var/www/ephemeral
+export CSM_PATH="${PITDATA}/csm-${CSM_RELEASE}"
+export SLS_INPUT_FILE=${PITDATA}/prep/${SYSTEM_NAME}/sls_input_file.json
+
+# Reset all the NCN keys
+truncate --size=0 /root/.ssh/known_hosts 2>&1
+grep -oP "(ncn-\w+)" /etc/hosts | sort -u | xargs -t -i ssh-keyscan -H \{\} >> /root/.ssh/known_hosts
+
+# Copy the k8s admin conf from m002
+scp ncn-m002:/etc/kubernetes/admin.conf /etc/kubernetes/admin.conf
+cp /etc/kubernetes/admin.conf $HOME/.kube/config
+
+# Clear the whole IPVS routing table on m001 (this may have been setup if kube-proxy was ever running on m001).
+# This will allow for m001 to use unbound once it is running in the k8s cluster.
+ipvsadm --clear
+
+# Start the CSM install
+pushd /usr/share/doc/csm/install/scripts/csm_services || err_exit "Failed changing directory to /usr/share/doc/csm/install/scripts/csm_services"
+yapl -f install1thru4.yaml execute --no-cache
+if [[ $? -ne 0 ]]; then
+  err_exit "Address any errors above and re-run rapid_reinstall.sh"
+fi
+
+# Edit the host file to move from the pit nexus to k8s nexus for packages.local and registry.local.
+# This is needed before step5 where k8s nexus is setup.
+UNBOUNDIP=10.92.100.71
+ncnregistry="${UNBOUNDIP}    registry.local packages.local"
+sed -i "s/.*registry.*/${ncnregistry}/" /etc/hosts
+
+# Complete the CSM install
+yapl -f install5thru7.yaml execute --no-cache
+popd || err_exit "Failed returning to previous directory"
+
+# Post install WARs
+# If the keycloak users-localize job failed, re-run it.
+kubectl get job -l app.kubernetes.io/instance=cray-keycloak-users-localize -n services --no-headers | grep -q "1/1"
+if [[ $? -eq 0 ]]; then
```

Comment:
This could also be simpified to one line:
```
if kubectl get job -l app.kubernetes.io/instance=cray-keycloak-users-localize -n services --no-headers | grep -q "1/1"; then
```

Reply:
fixed